src/components/NewCommentForm.tsx

```diff
@@ -1,8 +1,96 @@
-import React from 'react';
+import React, { useState } from 'react';
+import classNames from 'classnames';
+
+import { postComment } from '../api/data';
+
+type Props = {
+  postId: number
+  getNewComments: (postId: number) => Promise<void>;
+  commentError: (param: boolean) => void
+};
+
+export const NewCommentForm: React.FC<Props>
+= React.memo(({ postId, getNewComments, commentError }) => {
+  const [name, setName] = useState('');
+  const [email, setEmail] = useState('');
+  const [comment, setComment] = useState('');
+
+  const [isNameError, setIsNameError] = useState(false);
+  const [isEmailError, setIsEmailError] = useState(false);
+  const [isCommentError, setIsCommentError] = useState(false);
+  const [isCommentLoading, setIsCommentLoading] = useState(false);
+
+  const handleInput = (
+    event: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>,
+    callback: (param: string) => void,
+  ) => {
+    const { value } = event.currentTarget;
+
+    callback(value);
+  };
+
+  const handleClear = () => {
+    if (isEmailError) {
+      setIsEmailError(false);
+    }
+
+    if (isCommentError) {
+      setIsCommentError(false);
+    }
+
+    if (isNameError) {
+      setIsNameError(false);
+    }
```

Comment:
Do you really need to check the current value of errors? Just set 'false' for all of them

Reply:
I wanted to avoid unnecessary component redraw)